conftest.py

```diff
@@ -0,0 +1,266 @@
+    See https://docs.pytest.org/en/stable/reference.html#_pytest.hookspec.pytest_addoption
+    """     # noqa: E501
+    parser.addoption(
+        "--local", action="store_true", default=False, help="use local boards",
+    )
+    parser.addoption(
+        "--hide-output", action="store_true", default=False,
+        help="Don't log output of nodes",
+    )
+    parser.addoption(
+        "--boards", type=list_from_string,
+        help="list of BOARD's or IOTLAB_NODEs for the test",
+    )
+    parser.addoption(
+        "--self-test", action="store_true", default=False,
+        help="Tests the testutils rather than running the release tests",
+    )
+
+
+def pytest_ignore_collect(path, config):
+    """
+    return True to prevent considering this path for collection.
+
+    See: https://docs.pytest.org/en/stable/reference.html#_pytest.hookspec.pytest_ignore_collect
+    """     # noqa: E501
+    # This is about the --self-test option so I don't agree with pylint here
+    # pylint: disable=R1705
+    if config.getoption("--self-test"):
+        return "testutils" not in str(path)
+    else:
+        return "testutils" in str(path)
+
+
+def pytest_collection_modifyitems(config, items):
+    # pylint: disable=C0301
+    """
+    called after collection has been performed, may filter or re-order the
+    items in-place.
+
+    See: https://docs.pytest.org/en/stable/reference.html#_pytest.hookspec.pytest_collection_modifyitems
+    """     # noqa: E501
+    if os.geteuid() == 0:
+        # user is root, can run sudo tests
+        sudo_user = True
+    else:
+        sudo_user = False
+        sudo_only = pytest.mark.skip(reason="Test needs sudo to run")
+
+    if config.getoption("--local"):
+        # --local given in cli
+        run_local = True
+    else:
+        run_local = False
+        local_only = pytest.mark.skip(reason="Test can't run on IoT-LAB")
+
+    if not IoTLABExperiment.check_user_credentials() and not run_local:
+        no_credentials = True
+        iotlab_creds = pytest.mark.skip(
+            reason="Test requires IoT-LAB credentials in {}. "
+                   "Use `iotlab-auth` to create".format(
+                       os.path.join(os.environ["HOME"], ".iotlabrc"))
+        )
+    else:
+        no_credentials = False
+
+    for item in items:
+        if "local_only" in item.keywords and not run_local:
+            item.add_marker(local_only)
+        if "sudo_only" in item.keywords and not sudo_user:
+            item.add_marker(sudo_only)
+        if "iotlab_creds" in item.keywords and no_credentials:
+            item.add_marker(iotlab_creds)
+
+
+def pytest_keyboard_interrupt(excinfo):
+    # pylint: disable=C0301
+    """
+    Called on KeyInterrupt
+
+    See: https://docs.pytest.org/en/stable/reference.html?highlight=hooks#_pytest.hookspec.pytest_keyboard_interrupt
+    """     # noqa: E501
+    for child in RUNNING_CTRLS:
+        child.stop_term()
+    for exp in RUNNING_EXPERIMENTS:
+        exp.stop()
```

Comment:
Another thing that is flaky, even with this function, is cleanup. When the tests run uninterrupted, everything is cleaned up properly, but once one hits `Ctrl+C` during the tests some nodes and experiments might still be left running. Maybe someone with more experience with `pytest` can help here. @aabadie @fjmolinas?

Reply:
(it seems that both commands wait a bit after doing the actual stopping while pytest is too impatient and just kills the waiting).